jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/SelectableChannelEndPoint.java

```diff
@@ -115,6 +114,12 @@ public SelectableChannelEndPoint(Scheduler scheduler, SelectableChannel channel,
         _channel = channel;
         _selector = selector;
         _key = selectionKey;
+        /* Cache the local and remote Socket Address in this EndPoint class.
+         * as they are inaccessible once the EndPoint is closed.
+         * This helps RequestLog to access the remote and local addresses, even on a closed connection.
+         */
+        _localSocketAddress = getLocalSocketAddress();
+        _remoteSocketAddress = getRemoteSocketAddress();
```

Comment:
I guess it's possible to have a `ClosedChannelException` during the Constructor now.
Don't see that as a problem, but it's worth mentioning.